src/main/drivers/barometer/barometer_ms5611.c

```diff
@@ -45,45 +45,113 @@
 #define CMD_PROM_RD             0xA0 // Prom read command
 #define PROM_NB                 8
 
-static void ms5611_reset(void);
-static uint16_t ms5611_prom(int8_t coef_num);
+static void ms5611_reset(busDevice_t *pBusdev);
+static uint16_t ms5611_prom(busDevice_t *pBusdev, int8_t coef_num);
 STATIC_UNIT_TESTED int8_t ms5611_crc(uint16_t *prom);
-static uint32_t ms5611_read_adc(void);
-static void ms5611_start_ut(void);
-static void ms5611_get_ut(void);
-static void ms5611_start_up(void);
-static void ms5611_get_up(void);
+static uint32_t ms5611_read_adc(busDevice_t *pBusdev);
+static void ms5611_start_ut(baroDev_t *baro);
+static void ms5611_get_ut(baroDev_t *baro);
+static void ms5611_start_up(baroDev_t *baro);
+static void ms5611_get_up(baroDev_t *baro);
 STATIC_UNIT_TESTED void ms5611_calculate(int32_t *pressure, int32_t *temperature);
 
 STATIC_UNIT_TESTED uint32_t ms5611_ut;  // static result of temperature measurement
 STATIC_UNIT_TESTED uint32_t ms5611_up;  // static result of pressure measurement
 STATIC_UNIT_TESTED uint16_t ms5611_c[PROM_NB];  // on-chip ROM
 static uint8_t ms5611_osr = CMD_ADC_4096;
+static bool ms5611InitDone = false;
```

Comment:
I really don't like the use of this kind of flag. Instead the code should be written so that init is not called more than once.